05/app.js

```diff
@@ -13,29 +13,36 @@ Student.prototype.getAverageGrade = function (subject) {
 	let avgGradesArr = [];
 
 	// dla konkretnego przedmiotu
+
 	if (this.grades[subject] && this.grades[subject].length !== 0) {
-		const count = this.grades[subject].reduce((acc, curr) => acc + curr) / this.grades[subject].length;
-		return count;
+		const subjectAverage = countAverage(this.grades[subject], this.grades[subject].length);
+		return subjectAverage;
 	}
 
 	// dla ogolnej sredniej
 	if (!subject) {
 		for (const subjectName in this.grades) {
 			const grades = this.grades[subjectName];
-			const avgGrades = grades.reduce((acc, curr) => acc + curr) / grades.length;
-			avgGradesArr.push(avgGrades);
+			const mainAvgCounting = countAverage(grades, grades.length);
+			avgGradesArr.push(mainAvgCounting);
 		}
 	}
-	const mainAvg = avgGradesArr.reduce((acc, curr) => acc + curr) / avgGradesArr.length;
+	const mainAvg = countAverage(avgGradesArr, avgGradesArr.length);
 	return mainAvg;
 };
 
+function countAverage(subject, numberOfGrades) {
```

Comment:
ok, ale ta funkcja powinna być wew. `Student`, aby mieć wszystko pogrupowane w jednym miejscu i łatwy dostęp do danych :)

Reply:
zrobiłem to z myśla, ze ta funkcja moze byc uniwersalna bo moze liczyc srednia z czegokolwiek, moze te nazwy parametrów zdradzaja, ze chodzi tylko o przedmioty i powinny być metoda obiektu Student.
Dodane dodane jako metoda w prototype.

Reply:
Jasne, rozumiem :)